tests/functional/object/mcg/test_lifecycle_configuration.py

```diff
@@ -367,3 +376,396 @@ def test_expired_object_delete_marker(self, mcg_obj, bucket_factory, awscli_pod)
+
+    @tier2
+    @pytest.mark.polarion_id("OCS-6804")
+    def test_lifecycle_rules_combined(
+        self,
+        mcg_obj,
+        bucket_factory,
+        awscli_pod,
+        test_directory_setup,
+    ):
+        """
+        1. Create an MCG bucket with versioning enabled
+        2. Set onto the bucket a lifecycle configuration with:
+        Expiration.Days, Expiration.ExpiredObjectDeleteMarker
+        NoncurrentVersionExpiration.NoncurrentDays,
+        NoncurrentVersionExpiration.NewerNoncurrentVersions,
+        AbortIncompleteMultipartUpload.DaysAfterInitiation
+
+        3. Create a multipart-upload, upload a few parts.
+        4. Expire the multipart-upload and wait for it to expire
+        5. Upload 10 versions of the same object onto the bucket
+        6. Expire the non-current versions by setting back their creation date
+        7. Wait for the any non-current versions after the first 5 to get deleted
+        8. Expire the versioned object and wait for the delete marker to get created
+        9. Delete all the remaining non-delete-marker versions by their VersionId
+        10. Wait for the delete marker to get deleted
+        """
+        max_non_current_versions = 5
+
+        # 1. Create an MCG bucket with versioning enabled
+        bucket = bucket_factory(interface="OC")[0].name
+        put_bucket_versioning_via_awscli(mcg_obj, awscli_pod, bucket)
+
+        # 2. Set lifecycle configuration
+        lifecycle_policy = LifecyclePolicy(
+            ExpirationRule(days=5),
+            ExpiredObjectDeleteMarkerRule(),
+            NoncurrentVersionExpirationRule(
+                non_current_days=1, newer_non_current_versions=max_non_current_versions
+            ),
+            AbortIncompleteMultipartUploadRule(days_after_initiation=7),
+        )
+        mcg_obj.s3_client.put_bucket_lifecycle_configuration(
+            Bucket=bucket, LifecycleConfiguration=lifecycle_policy.as_dict()
+        )
+        logger.info(
+            f"Sleeping for {PROP_SLEEP_TIME} seconds to let the policy propagate"
+        )
+        sleep(PROP_SLEEP_TIME)
+
+        # 3. Create a multipart-upload, upload a few parts.
+        parts_amount = 5
+        origin_dir = test_directory_setup.origin_dir
+        res_dir = test_directory_setup.result_dir
+
+        multipart_upload_obj_key = "multipart_object"
+        upload_id = create_multipart_upload(mcg_obj, bucket, multipart_upload_obj_key)
+        awscli_pod.exec_cmd_on_pod(
+            f'sh -c "dd if=/dev/urandom of={origin_dir}/{multipart_upload_obj_key} bs=1MB count={parts_amount}; '
+            f'split -b 1m  {origin_dir}/{multipart_upload_obj_key} {res_dir}/part"'
+        )
+        parts = awscli_pod.exec_cmd_on_pod(f'sh -c "ls -1 {res_dir}"').split()
+        upload_parts(
+            mcg_obj,
+            awscli_pod,
+            bucket,
+            multipart_upload_obj_key,
+            res_dir,
+            upload_id,
+            parts,
+        )
```

Comment:
@sagihirshfeld As per step3, script should upload some parts and not all. Are we uploading few parts or all parts here?

Reply:
You're correct - I'm actually uploading all the parts and not just a few, so I'll correct the docstring and comment.

For testing the AbortIncompleteMultipartUpload rule though, it doesn't really matter whether all the parts are available or not. As long as we didn't finalize the upload with CompleteMultipartUpload command, it should still expire if it lingered long enough.

Reply:
Fixed the docstring and comment here: bc08ba969e805cf5f62f47062e4acb16808a3a86